udev/udev-configure-printer.c

```diff
@@ -1830,8 +1863,8 @@ do_remove (const char *devaddr)
   map = read_usb_uri_map ();
   prev = &map->entries;
   for (entry = map->entries; entry; entry = entry->next)
-    {
-      if (!strcmp (entry->devpath, devpath))
+  {
+      if (compare_usb_uri(entry->devpath, devpath))
```

Comment:
What about using ```strstr()``` instead of ```strcmp()```. It will do the trick with finding an exact match and substring too.

Then you can add ```device_exists()``` to check if the device really exists and move the code I marked into it.